Sources/Prometheus/Histogram.swift

```diff
@@ -0,0 +1,174 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+public protocol Bucketable: AdditiveArithmetic, Comparable {
```

Comment:
Does this need to be `public`? I guess in general, do we expect people to use all those types without going through `swift-metrics`?

Reply:
Yes, I think there might be use-cases outside using this as a backend for `swift-metrics`. We can make this non-public but then we must provide two explicit Histogram implementations.

Reply:
Fine by me but a bit of documentation wouldn't hurt on this since it is something that is in addition to the normal `swift-metrics` stuff.